pkg/linode-bs/nodeserver.go

```diff
@@ -236,11 +280,37 @@ func (ns *LinodeNodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeSt
 		return nil, status.Error(codes.Unimplemented, "Block volume support is not yet implemented")
 	}
 
-	err = ns.Mounter.FormatAndMount(devicePath, stagingTargetPath, fstype, options)
-	if err != nil {
+	glog.V(4).Info("luksContext encrypted ", luksContext.EncryptionEnabled)
+	formatted, err := blkidValid(devicePath)
+	glog.V(4).Info("luks volume formatted  ", formatted)
+	if (!formatted && luksContext.EncryptionEnabled) {
```

Comment:
need to clean up the conditionals a bit, no need for two encryption enabled checks etc.

Reply:
thinking about this a bit more, i'd check if encryption is enalbed before even calling `getLuksContext` or `blkidValid`... none of it should be called if we're not even using encryption